sdk/resourcemanager/azure-resourcemanager-keyvault/src/main/java/com/azure/resourcemanager/keyvault/implementation/SecretsImpl.java

```diff
@@ -110,7 +123,51 @@ public Secret getByNameAndVersion(String name, String version) {
 
     @Override
     public Mono<Secret> getByNameAndVersionAsync(final String name, final String version) {
-        return inner.getSecret(name, version).map(this::wrapModel);
+        Objects.requireNonNull(name);
+        return (version == null ? inner.getSecret(name) : inner.getSecret(name, version)).map(this::wrapModel);
+    }
+
+    @Override
+    public Secret enableByNameAndVersion(String name, String version) {
+        return enableByNameAndVersionAsync(name, version).block();
+    }
+
+    @Override
+    public Mono<Secret> enableByNameAndVersionAsync(String name, String version) {
+        Objects.requireNonNull(name);
+        return enableByNameAndVersionAsync(name, version, true);
+    }
+
+    @Override
+    public void disableByNameAndVersion(String name, String version) {
+        disableByNameAndVersionAsync(name, version).block();
+    }
+
+    @Override
+    public Mono<Void> disableByNameAndVersionAsync(String name, String version) {
+        Objects.requireNonNull(name);
+        return enableByNameAndVersionAsync(name, version, false).then();
+    }
+
+    private Mono<Secret> enableByNameAndVersionAsync(String name, String version, boolean enabled) {
+        try {
+            String mockId = "https://foo.vault.azure.net/secrets/" + name;
```

Comment:
Is it any other way rather than using such a mock ID?

Reply:
The problem is that `SecretProperties` does not have constructor that can take the `version`. So the only way I can create a `SecretProperties` with `version` is to use serialization (https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/keyvault/azure-security-keyvault-secrets/src/main/java/com/azure/security/keyvault/secrets/models/SecretProperties.java#L330-L344).

Reply:
Alternative would to do a `list`, then choose the correct `SecretProperties`.

We cannot do `get`, since a disabled secret would fail in `get`.